bin/sync_django_ldap.py

```diff
@@ -62,35 +62,27 @@ def main():
     # Creating this here because this is a singleton class
     _ = LdapQuery(VscConfiguration(VSC_CONF_DEFAULT_FILENAME))
 
-    last_timestamp = opts.options.start_timestamp
-    if not last_timestamp:
-        try:
-            last_timestamp = read_timestamp(SYNC_TIMESTAMP_FILENAME)
-        except Exception:
-            _log.warning("Something broke reading the timestamp from %s", SYNC_TIMESTAMP_FILENAME)
-            last_timestamp = "201710230000Z"
-            _log.warning("We will resync from a hardcoded know working sync a while back : %s", last_timestamp)
-
-    _log.info("Using timestamp %s", last_timestamp)
-    # record starttime before starting, and take a 10 sec safety buffer so we don't get gaps where users are approved
-    # in between the requesting of modified users and writing out the start time
-    start_time = datetime.datetime.now(tz=utc) + datetime.timedelta(seconds=-10)
-    _log.info("startime %s", start_time)
+    (last_timestamp, start_time) = retrieve_timestamp_with_default(
+        SYNC_TIMESTAMP_FILENAME,
+        opts.options.start_timestamp,
+        "201710230000Z")
+    logging.info("Using timestamp %s", last_timestamp)
+    logging.info("Using startime %s", start_time)
 
     try:
         parent_pid = os.fork()
-        _log.info("Forked.")
+        logging.info("Forked.")
     except OSError:
-        _log.exception("Could not fork")
+        logging.exception("Could not fork")
         parent_pid = 1
     except Exception:
-        _log.exception("Oops")
+        logging.exception("Oops")
         parent_pid = 1
 
     if parent_pid == 0:
         try:
-            global _log
-            _log = fancylogger.getLogger(NAGIOS_HEADER)
+            global logger
+            logger = fancylogger.getLogger(NAGIOS_HEADER)
```

Comment:
why change it here and not at the top?

Reply:
probably because child and parent don't mess up the logging.
the better question is : why is there any forking at all in this sync script